doc/decisions/warning-array.md

```diff
@@ -0,0 +1,36 @@
+# Arrays
+
+## Problem
+
+Currently multiple warnings are saved in an elektra non-conforming array
+notation which is limited to 100 entries. The notation of `#00` is against
+the design [decision made](array.md).
+
+Furthermore if the number of warnings is exceeded, the earliest entries are overwritten
+which might yield the most useful warning messages.
+
+## Constraints
+
+- The number of entries should stay limited but may be more than 100 entries.
+
+## Assumptions
+
+## Considered Alternatives
+
+## Decision
+
+The format should be aligned with the correct array notation,
+starting with `#0`. The maximum number of warnings will be expanded to
+1001 entries (`#0` - `#1000`).
+
+If more warnings occur, warning `#1000` is overwritten each time.
+
+## Rationale
```

Comment:
Missing. (To confirm to the array notation.)

Is there any reason why it should be 1001?

Reply:
1001 is just because it is more natural (to me) to run from 0-1000 instead of 0-999. I have no problem to limit it somewhere else